src/tests.rs

```diff
@@ -414,3 +414,96 @@ fn it_should_display_hex() {
         "00000000\t61 62 63    \tabc\n".to_owned()
     );
 }
+
+#[test]
+fn line_of_empty_span_is_empty() {
+    assert_eq!(
+        StrSpan::new("").get_line(),
+        Some("".as_ref()),
+    );
+}
+
+#[test]
+fn line_of_single_line_start_is_whole() {
+    assert_eq!(
+        StrSpan::new("A single line").get_line(),
+        Some("A single line".as_ref()),
+    );
+}
+#[test]
+fn line_of_single_line_end_is_whole() {
+    let data = "A single line";
+    assert_eq!(
+        StrSpan::new(data).slice(data.len()..).get_line(),
+        Some("A single line".as_ref()),
+    );
+}
+
+#[test]
+fn line_of_start_is_first() {
+    assert_eq!(
+        StrSpan::new(
+            "One line of text\
+             \nFollowed by a second\
+             \nand a third\n"
+        ).get_line(),
+        Some("One line of text".as_ref()),
+    );
+}
+
+#[test]
+fn line_of_nl_is_before() {
+    let data =
+        "One line of text\
+         \nFollowed by a second\
+         \nand a third\n";
+    assert_eq!(
+        StrSpan::new(data).slice(data.find('\n').unwrap()..).get_line(),
+        Some("One line of text".as_ref()),
+    );
+}
+
+#[test]
+fn line_of_end_after_nl_is_empty() {
+    let data =
+        "One line of text\
+         \nFollowed by a second\
+         \nand a third\n";
+    assert_eq!(
+        StrSpan::new(data).slice(data.len()..).get_line(),
+        Some("".as_ref()),
+    );
+}
+
+#[test]
+fn line_of_end_no_nl_is_last() {
+    let data =
+        "One line of text\
+         \nFollowed by a second\
+         \nand a third";
+    assert_eq!(
+        StrSpan::new(data).slice(data.len()..).get_line(),
+        Some("and a third".as_ref()),
+    );
+}
+
+#[test]
+fn line_for_non_ascii_chars() {
+    // I don't really know if this Oriya text makes sense.
```

Comment:
I don't think that's Oriya :p

Reply:
Nope.  I decided to change it to a language I do know (my native Swedish), but forgot the comment.  :-)